engine/src/main/java/org/hibernate/validator/spi/nodenameprovider/PropertyNodeNameProvider.java

```diff
@@ -0,0 +1,11 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.spi.nodenameprovider;
+
+public interface PropertyNodeNameProvider {
+	String getName(String propertyName, Property property);
```

Comment:
I think we should just have
```java
String getName(Property property);
```
And property interface would be:
```java
public interface Property {
	String getName();
}
```
property itself should already contain information about its own name.